CONTRIBUTING.md

```diff
@@ -0,0 +1,236 @@
+# Contributing to rust-bitcoin
+
+:+1::tada: First off, thanks for taking the time to contribute! :tada::+1:
+
+The following is a set of guidelines for contributing to Rust Bitcoin
+implementation and other Rust Bitcoin-related projects, which are hosted in the
+[Rust Bitcoin Community](https://github.com/rust-bitcoin) on GitHub. These are
+mostly guidelines, not rules. Use your best judgment, and feel free to propose
+changes to this document in a pull request.
+
+#### Table Of Contents
+
+- [General](#general)
+- [Communication channels](#communication-channels)
+- [Asking questions](#asking-questions)
+- [Contribution workflow](#contribution-workflow)
+  * [Preparing PRs](#preparing-prs)
+  * [Peer review](#peer-review)
+  * [Repository maintainers](#repository-maintainers)
+- [Coding conventions](#coding-conventions)
+  * [Formatting](#formatting)
+  * [MSRV](#msrv)
+  * [Naming conventions](#naming-conventions)
+  * [Unsafe code](#unsafe-code)
+- [Security](#security)
+- [Testing](#testing)
+- [Going further](#going-further)
+
+
+## General
+
+The Rust Bitcoin project operates an open contributor model where anyone is
+welcome to contribute towards development in the form of peer review,
+documentation, testing and patches.
+
+Anyone is invited to contribute without regard to technical experience,
+"expertise", OSS experience, age, or other concern. However, the development of
+standards & reference implementations demands a high-level of rigor, adversarial
+thinking, thorough testing and risk-minimization. Any bug may cost users real
+money. That being said, we deeply welcome people contributing for the first time
+to an open source project or pick up Rust while contributing. Don't be shy,
+you'll learn.
+
+
+## Communication channels
+
+Communication about Rust Bitcoin happens primarily in
+[#bitcoin-rust](https://web.libera.chat/?channel=#bitcoin-rust) IRC chat on
+[Libera](https://libera.chat/) with the logs available at
+<http://gnusha.org/rust-bitcoin/>.
```

Comment:
It looks the logs at http://gnusha.org/rust-bitcoin/ didn't switch to libera because they are not working since about June 2021

Reply:
Yes. We have to contact Bryan Bishop to do that.

Reply:
Sure, I'll ping him.

Reply:
Actually I just checked and it looks liek https://gnusha.org/bitcoin-rust/ is up to date.

So actually, we should change our CONTRIBUTING file and our IRC /topic.

Reply:
May I do that with a fixup commit not to make re-review the whole PR for others?

Reply:
I vote for fixup, reviewing this is a huge pain already.

Reply:
Yep for sure.

Reply:
> Actually I just checked and it looks liek https://gnusha.org/bitcoin-rust/ is up to date.

Is it correct that days are there, but they are empty (except from log-start/log-end) up to June 2021?

Reply:
There is very little activity on the channel, but the logs do not appear empty to me.

Reply:
Ah sorry I was watching http://gnusha.org/rust-bitcoin/ while you linked http://gnusha.org/bitcoin-rust/ (rust and bitcoin switched)

Reply:
Done